Makefile

```diff
@@ -257,3 +263,35 @@ crd-controller-deploy:
 .PHONY: crd-controller-test
 crd-controller-test:
 	$(CRD_MAKE) manifests generate install test
+
+.PHONY: envtest
+envtest: $(ENVTEST) ## Download envtest-setup locally if necessary.
+$(ENVTEST): $(BIN)
+	test -s $(BIN)/setup-envtest || GOBIN=$(CUR_DIR)/$(BIN) go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
+
+.PHONY: ginkgo
+ginkgo: $(GINKGO) ## Download ginkgo locally if necessary.
+$(GINKGO): $(BIN)
+	test -s $(BIN)/ginkgo || GOBIN=$(CUR_DIR)/$(BIN) go install github.com/onsi/ginkgo/v2/ginkgo@latest
+
+.PHONY: minio-install
+minio-install:
+	echo "Install Minio server"
+	helm repo add bitnami https://charts.bitnami.com/bitnami
+	helm repo update
+	helm upgrade -i minio --namespace=minio bitnami/minio --create-namespace \
+ 		--set defaultBuckets=integrity \
+ 		--wait
+
+.PHONY: load-images
+load-images:
+	minikube image load $(FULL_IMAGE_NAME) controller:latest
+
+.PHONY: install-test-cr
+install-test-cr:
+	@helm upgrade -i test-cr e2e/chart/snapshot --wait
+
+.PHONY: e2etest
+e2etest: install-test-cr envtest ginkgo
```

Comment:
Probably the order of targets should be different.

Reply:
So, why? Is there the reason?